pyanaconda/kickstart.py

```diff
@@ -1915,6 +1914,70 @@ def execute(self, *args):
             desktop.runlevel = 3
             desktop.write()
 
+class Snapshot(commands.snapshot.RHEL7_Snapshot):
+    def setup(self, storage, ksdata, instClass):
+        """ Prepare the snapshot.
+
+            This will also do the checking of snapshot validity.
+        """
+        for snap_data in self.dataList():
+            snap_data.setup(storage, ksdata, instClass)
+
+    def execute(self, storage, ksdata, instClass):
+        """ Create ThinLV snapshot.
+
+            The Blivet must be reseted before creation of the snapshot. This is
+            required because the storage could be changed in post section.
+        """
+        try_populate_devicetree(storage.devicetree)
```

Comment:
I think I had this problem with re-creation failing so I changed it here to this. Maybe it won't be able to catch everything in post but it's better than nothing.